common/src/test/java/tachyon/security/authentication/AuthenticationFactoryTest.java

```diff
@@ -22,49 +43,387 @@
+
     // should return a SIMPLE AuthType with conf "SIMPLE"
-    tachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "SIMPLE");
-    authType = AuthenticationFactory.getAuthTypeFromConf(tachyonConf);
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "SIMPLE");
+    authType = new AuthenticationFactory(mTachyonConf).getAuthType();
     Assert.assertEquals(AuthenticationFactory.AuthType.SIMPLE, authType);
 
-    // should return a SIMPLE AuthType with conf "simple"
-    tachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "simple");
-    authType = AuthenticationFactory.getAuthTypeFromConf(tachyonConf);
+    // should return a CUSTOM AuthType with conf "CUSTOM"
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "CUSTOM");
+    authType = new AuthenticationFactory(mTachyonConf).getAuthType();
+    Assert.assertEquals(AuthenticationFactory.AuthType.CUSTOM, authType);
+
+    // should return a KERBEROS AuthType with conf "KERBEROS"
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "KERBEROS");
+    authType = new AuthenticationFactory(mTachyonConf).getAuthType();
+    Assert.assertEquals(AuthenticationFactory.AuthType.KERBEROS, authType);
+
+    // case insensitive - should return a SIMPLE AuthType with conf "simple"
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "simple");
+    authType = new AuthenticationFactory(mTachyonConf).getAuthType();
     Assert.assertEquals(AuthenticationFactory.AuthType.SIMPLE, authType);
 
-    // should throw exception with conf "wrong"
-    tachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "wrong");
+    // wrong configuration - should throw exception with conf "wrong"
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "wrong");
     mThrown.expect(IllegalArgumentException.class);
     mThrown.expectMessage("wrong is not a valid authentication type. Check the configuration "
         + "parameter " + Constants.TACHYON_SECURITY_AUTHENTICATION);
-    authType = AuthenticationFactory.getAuthTypeFromConf(tachyonConf);
+    authType = new AuthenticationFactory(mTachyonConf).getAuthType();
+  }
+
+  /**
+   * In NOSASL mode, the TTransport used should be the same as Tachyon original code.
+   */
+  @Test
+  public void nosaslAuthenticationTest() throws Exception {
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "NOSASL");
+
+    // start server
+    startServerThread(mTachyonConf);
+
+    // create client and connect to server
+    TTransport client = createClient(mTachyonConf, null, null);
+    client.open();
+    Assert.assertTrue(client.isOpen());
+
+    // clean up
+    client.close();
+    mServer.stop();
+  }
+
+  /**
+   * In SIMPLE mode, the TTransport mechanism is PLAIN. When server authenticate the connected
+   * client user, it use {@link tachyon.security.authentication.SimpleAuthenticationProviderImpl}.
+   */
+  @Test
+  public void simpleAuthenticationTest() throws Exception {
```

Comment:
Personally, I think `null` check and empty string check for username and password should not be done by the authorization provider. Because authorization can be implemented by any developers and they miss the check, it's left for the server code to deal with it.